controllers/argocd/namespaceManagement_test.go

```diff
@@ -0,0 +1,189 @@
+package argocd
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	argoproj "github.com/argoproj-labs/argocd-operator/api/v1beta1"
+	"github.com/argoproj-labs/argocd-operator/common"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	testclient "k8s.io/client-go/kubernetes/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
```

Comment:
Can you add additional test cases for testing `reconcileNamespaceManagement`? This test is important to maintaining security of this feature, so it would be good to ensure that all of the things this functions is doing are tested.